exercises/rail-fence-cipher/example.rb

```diff
@@ -0,0 +1,72 @@
+# rubocop:enable all
+class RailFenceCipher
+  VERSION = 1
+
+  def self.encode(message, rails)
+    return message if message.empty? || rails == 1
+    empty_fences = create_empty_fences_array(message, rails)
+    zig_zag_array = populate_fences_array(message, rails, empty_fences)
+    zig_zag_array.flatten.join
+  end
+
+  def self.decode(message, rails)
+    return message if message.empty? || rails == 1
+    empty = create_empty_fences_array(message, rails)
+    mark_fences = populate_fences_array('X' * message.length, rails, empty)
+    line_lengths = find_the_lengths_of_the_lines(message, rails)
+    lines = split_the_encrypted_message_by_rows(message, line_lengths)
+    result = switch_marked_cells_with_string(mark_fences, lines, empty)
+    result.transpose.flatten.join
+  end
+
+  class << self
+    private
+
+    def create_empty_fences_array(message, rails)
+      Array.new(rails) { Array.new(message.length) { nil } }
+    end
+
+    def populate_fences_array(message, rails, fences)
+      row = 0
+      row_direction = 1
+      (0..(message.length - 1)).each do |column|
```

Comment:
We could use `...` here, meaning that you want to stop right before the `message.length`.

Another possibility is to use the `times` method, in this case it would be  `message.length.times do |column|`